lib/environment_config_transmogrifier.rb

```diff
@@ -75,7 +75,7 @@ def self.inject_value(hash, key_grams, value)
     # error out if we're trying to override
     # an existing value that's not a hash
     unless hash[key_grams[0]].is_a?(Hash)
-      raise NonHashValueOverride, 'Refusing to override non-hash value.'
+      raise NonHashValueOverride, "Refusing to override non-hash value: #{key_grams}"
```

Comment:
This being a recursive function, it'll only log the conflicting part; it definitely helps, but will still be hard to locate. Do we want to make it hold on to the original too?